src/realmd/AuthSocket.cpp

```diff
@@ -613,7 +626,26 @@ bool AuthSocket::_HandleLogonProof()
                 });
                 return;
             }
+            bool pinResult = true;
+
+            if (self->m_promptPin && (lp->securityFlags & SECURITY_FLAG_PIN))
```

Comment:
Why even do this pin variable shenanigans. Just put the whole thing in an IF statement and check if token is empty there. Structure it like the if (lp->securityFlags & SECURITY_FLAG_AUTHENTICATOR || !self->_token.empty()) statement. In fact m_promptPin is redundant and only SECURITY_FLAG_PIN and token needs to be checked.

Reply:
there are multiple cases:
pin requested by the server and correct pin provided by the client
pin requested by the server and incorrect pin provided by the client
pin not requested by the server and pin provided by the client
so I use m_promptPin to check if server requested pin and SECURITY_FLAG_PIN to check if the client provided a PIN

Reply:
Well but we only care about if token is empty or pin was sent. I get that the other cases exist, but we do not need to handle those.